independent-projects/arc/processor/src/main/java/io/quarkus/arc/processor/BeanInfo.java

```diff
@@ -434,7 +438,14 @@ protected String getType() {
                 }
             }
 
-            Methods.addInterceptedMethodCandidates(beanDeployment, target.get().asClass(), candidates, classLevelBindings);
+            Set<MethodInfo> finalMethods = Methods.addInterceptedMethodCandidates(beanDeployment, target.get().asClass(),
+                    candidates, classLevelBindings);
+            if (!finalMethods.isEmpty()) {
+                errors.add(new DeploymentException(String.format(
+                        "Bean %s has a bound interceptor and must not declare final methods:\n\t- %s", getBeanClass(),
+                        finalMethods.stream().map(Object::toString).collect(Collectors.joining("\n\t- ")))));
```

Comment:
Minor nitpick but could you add a `sorted()` after the `map()` call? Having a deterministic order helps when you go through the error message to fix the issue.